File: atak/ATAKMapEngine/lib/src/main/java/com/atakmap/map/layer/raster/ImageInfo.java
package com.atakmap.map.layer.raster;

import com.atakmap.coremap.maps.coords.GeoPoint;
import com.atakmap.lang.Objects;

public class ImageInfo {
    public final String path;
    public final String type;
    public final boolean precisionImagery;
    public final GeoPoint upperLeft;
    public final GeoPoint upperRight;
    public final GeoPoint lowerRight;
    public final GeoPoint lowerLeft;
    public final double maxGsd;
    public final int width;
    public final int height;
    public final int srid;

    public ImageInfo(String path,
                     String type,
                     boolean precisionImagery,
                     GeoPoint upperLeft,
                     GeoPoint upperRight,
                     GeoPoint lowerRight,
                     GeoPoint lowerLeft,
                     double maxGsd,
                     int width,
                     int height,
                     int srid) {

        this.path = path;
        this.type = type;
        this.precisionImagery = precisionImagery;
        this.upperLeft = upperLeft;
        this.upperRight = upperRight;
        this.lowerRight = lowerRight;
        this.lowerLeft = lowerLeft;
        this.maxGsd = maxGsd;
        this.width = width;
        this.height = height;
        this.srid = srid;
    }

    @Override
    public int hashCode() {
        return this.path.hashCode();
    }

    @Override
    public boolean equals(Object o) {
        if(!(o instanceof ImageInfo))
            return false;
        return Objects.equals(this.path, ((ImageInfo)o).path);
    }
}

File: takengine/mapengine/sdk/src/raster/ImageInfo.cpp
#include "raster/ImageInfo.h"

using namespace TAK::Engine::Raster;

using namespace TAK::Engine::Core;

ImageInfo::ImageInfo() NOTHROWS :
    precisionImagery(false),
    maxGsd(0.0),
    width(0),
    height(0),
    srid(-1)
{}
ImageInfo::ImageInfo(const char *path_,
                     const char *type_,
                     const bool precisionImagery_,
                     const GeoPoint2 &upperLeft_,
                     const GeoPoint2 &upperRight_,
                     const GeoPoint2 &lowerRight_,
                     const GeoPoint2 &lowerLeft_,
                     const double maxGsd_,
                     const int width_,
                     const int height_,
                     const int srid_) NOTHROWS :
    path(path_),
    type(type_),
    precisionImagery(precisionImagery_),
    upperLeft(upperLeft_),
    upperRight(upperRight_),
    lowerRight(lowerRight_),
    lowerLeft(lowerLeft_),
    maxGsd(maxGsd_),
    width(width_),
    height(height_),
    srid(srid_)
{}

ImageInfo::~ImageInfo() NOTHROWS
{ }

ImageInfo &ImageInfo::operator=(const ImageInfo &) = default;